### DISCUSSION TITLE
Media player upgrade : QMediaPlayer

### DISCUSSION THREAD

**Discussion by ed2050** [zauberzeug/nicegui#1995] [Ideas / Feature Requests]
As mentioned in #1673, there's a quasar component [qmediaplayer](https://github.com/quasarframework/quasar-ui-qmediaplayer) with richer features than nicegui's current audio/video components.  It supports multiple tracks, language translations, keyboard shortcuts (native controls), mobile players, etc.  Would be nice to add to nicegui if possible.

Here's a link to a [live demo](https://quasarframework.github.io/quasar-ui-qmediaplayer/demo) of the current release version.  There's also a beta of the [2.0 release](https://qmediaplayer.netlify.app/all-about-qmediaplayer/what-is-qmediaplayer).

According to [docs](https://quasarframework.github.io/quasar-ui-qmediaplayer/docs), can install as a Quasar app extension with: `quasar ext add @quasar/qmediaplayer` .  
[More install info here](https://qmediaplayer.netlify.app/all-about-qmediaplayer/installation-types#App-Extension) (beta site).

## Status
Despite having a separate github page, qmediaplayer is closely connected to quasar.  
The [quasar video page](https://quasar.dev/vue-components/video) links to it, saying this:

> [!NOTE]
> You may also want to check our own HTML 5 video player component: [QMediaPlayer](https://github.com/quasarframework/app-extension-qmediaplayer), which is far more advanced than QVideo

I also note this from qmediaplayer's github page:
> If you appreciate the work that went into this, please consider donating to [Quasar](https://donate.quasar.dev/) or [Jeff](https://github.com/sponsors/hawkeye64).
> MIT (c) Jeff Galbraith [jeff@quasar.dev](mailto:jeff@quasar.dev)

So it's from a quasar dev, donations are directed to quasar itself, and the offical quasar documentation suggests using qmediaplayer.  Don't know why it's not included in quasar already; maybe they plan to merge it in the future.

**Comment by falkoschindler** [comment]
Sounds great, @ed2050!

Do you happen to know how to integrate a Quasar extension via UMD like we do with the core library?
```html
<script src="{{ prefix | safe }}/_nicegui/{{version}}/static/quasar.umd.prod.js"></script>
```

**Reply by ed2050** [reply]
Not yet.  Any docs on the process?

**Reply by falkoschindler** [reply]
Not really. I guess we could add the dependency to npm.json, run npm.py and use it like other third-party libraries when subclassing `ui.element`.
